recipes/openassetio/all/conanfile.py

```diff
@@ -0,0 +1,177 @@
+import pathlib
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.apple import is_apple_os
+from conan.tools.files import apply_conandata_patches, get, copy, rm
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "openassetio"
+    description = "An open-source interoperability standard for tools and content management systems used in media production."
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/OpenAssetIO/OpenAssetIO"
+    topics = ("asset-pipeline", "vfx", "cg", "assetmanager", "vfx-pipeline")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
```

Comment:
`fPIC` is a bit tricky in this case, since even if `shared=False` we still build a shared library for the Python extension module, which links in the core library. Thus the core library needs `fPIC=True` in all cases.

As an aside, the practical build matrix is a bit more complex, since it raises ODR questions if you link a static library build and also want to load the Python extension module into the same process.

Reply:
fair enough!